lib/3scale_toolbox/cli.rb

```diff
@@ -12,9 +14,137 @@ def self.load_builtin_commands
   end
 
   def self.run(args)
-    load_builtin_commands
-    ThreeScaleToolbox.load_plugins
-    # TODO handle error
-    root_command.build_command.run args
+    error_watchdog do
+      load_builtin_commands
+      ThreeScaleToolbox.load_plugins
+      root_command.build_command.run args
+    end
+  end
+
+  def self.error_watchdog(exit_on_error: true)
+    new.error_watchdog(exit_on_error: exit_on_error) { yield }
+  end
+
+  # Catches errors and prints nice diagnostic messages
+  def error_watchdog(exit_on_error:)
+    # Set exit handler
+    %w[INT TERM].each do |signal|
```

Comment:
This is just moving this from one module from another. It does not address the fact that this is global action that is for example not reverted back.

IMO this class does not have one reason to change. It does error handling but also installs signal handlers for an interruption. Those are two different concepts that should be separate.

`error_watchdog` should basically just be `yield` + `rescue`. Installing signal handlers should be done in `self.run` before entering the `error_watchdog`.